mesa/batchrunner.py

```diff
@@ -118,7 +118,7 @@ def collect_model_vars(self, model):
     def collect_agent_vars(self, model):
         """ Run reporters and collect agent-level variables. """
         agent_vars = {}
-        for agent in model.schedule.agents:
+        for agent in model.schedule.agents.values():
```

Comment:
IIRC the ordering is non-deterministic in `python<3.6`. Yet deterministic execution is crucial in scientific simulations.